CMakeLists.txt

```diff
@@ -242,6 +262,48 @@ add_compile_options(
     "$<$<BOOL:${ENABLE_TSAN}>:-fsanitize=thread>"
     "$<$<BOOL:${ENABLE_UBSAN}>:-fsanitize=undefined>"
 )
+
+else()
+
+add_compile_options(
```

Comment:
If my eyes are right, this entire list is a duplicate of the above with the exception of removing `-mavx2` as that is x86 specific.  Right?

Please just wrap the avx2 line as this:

```cmake
    $<$<STREQUAL:$<LOWER_CASE:$<TARGET_PROPERTY:PROCESSOR>>,x86_64>,-mavx2>
```

(untested: I might have a typo there)

Reply:
will do; wasn't sure how ya'll would want to see that change thank you for looking that over and providing feedback.